pkg/cli/image/extract/extract.go

```diff
@@ -357,7 +371,29 @@ func (o *ExtractOptions) Run() error {
 					return err
 				}
 
-				imageConfig, layers, err := imagemanifest.ManifestToImageConfig(ctx, srcManifest, repo.Blobs(ctx), location)
+				manifests, err := repo.Manifests(ctx)
+				if err != nil {
+					return err
+				}
+
+				// If empty, platformSpec will be set to match other mappings
+				// platformSpec will be used to determine which manifest from a DeserializedManifestList should be chosen, when appropriate.
```

Comment:
I'm confused - why is this being defaulted in extract?  If this is about oc adm release new, you need to calculate what platform you are looking for in oc adm release new, then pass down the more specific filter to all the sub commands (not interpret the logic here in extract). 

Only oc adm release new knows about releases, so that's the only code that can translate "release image + initial filterByOS" into "release image + fully qualified filterByOS"

Reply:
the platformSpec is found in `oc adm release new` but then has to be passed to extract->manifest.go bc that's where all the filtering/choosing happens.  Also, if _not_ in `oc adm release new`  it's still useful to pass the 'filterByOS' from a user to get better error msg if the given os/arch can't be matched to an available manifest in a manifestLIst.